test/e2e/runner/cli_provisioner.go

```diff
@@ -76,17 +76,22 @@ func (cli *CLIProvisioner) Run() error {
 
 func (cli *CLIProvisioner) provision() error {
 	cli.Config.Name = cli.generateName()
+	if cli.Config.SoakClusterName != "" {
+		cli.Config.Name = cli.Config.SoakClusterName
+	}
 	os.Setenv("NAME", cli.Config.Name)
 	log.Printf("Cluster name:%s\n", cli.Config.Name)
 
 	outputPath := filepath.Join(cli.Config.CurrentWorkingDir, "_output")
 	os.Mkdir(outputPath, 0755)
 
-	out, err := exec.Command("ssh-keygen", "-f", cli.Config.GetSSHKeyPath(), "-q", "-N", "", "-b", "2048", "-t", "rsa").CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("Error while trying to generate ssh key:%s\nOutput:%s", err, out)
+	if cli.Config.SoakClusterName == "" {
+		out, err := exec.Command("ssh-keygen", "-f", cli.Config.GetSSHKeyPath(), "-q", "-N", "", "-b", "2048", "-t", "rsa").CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("Error while trying to generate ssh key:%s\nOutput:%s", err, out)
+		}
+		exec.Command("chmod", "0600", cli.Config.GetSSHKeyPath()+"*")
 	}
-	exec.Command("chmod", "0600", cli.Config.GetSSHKeyPath()+"*")
 
 	publicSSHKey, err := cli.Config.ReadPublicSSHKey()
```

Comment:
if the above keygen is skipped when soak cluster name is provided, should the publicSSHKey part be skipped too?

Reply:
The keygen is skipped because for Jenkins soak test the tests use the Jenkins key so it is not necessary to generate a new key. However, the cli provisioner still needs to read the public key that was provided.

Reply:
Just to confirm, so the public key is not the one generated by keygen, right?

Reply:
No, it's the one in _output/cluster-name-ssh which is created before the test is ran

Reply:
Thanks for the clarification.